lib/main/permissions/notification_permission_service.dart

```diff
@@ -0,0 +1,49 @@
+import 'dart:io';
+
+import 'package:permission_handler/permission_handler.dart';
+import 'package:tmail_ui_user/features/base/isolate/background_isolate_binary_messenger/background_isolate_binary_messenger_mobile.dart';
+import 'package:tmail_ui_user/main/permissions/permission_service.dart';
+
+class NotificationPermissionService extends PermissionService {
+  String _notificationGroupId = '';
+
+  @override
+  Future<bool> isGranted(Permission permission) async {
+    if (permission.value != Permission.notification.value) {
+      return callSuperIsGranted(permission);
+    }
+
+    if (!Platform.isAndroid) {
+      return callSuperIsGranted(permission);
+    }
+
+    final current = await permission.status;
+    switch (current) {
+      case PermissionStatus.granted:
+        return checkGroupPermission();
+      default:
+        return callSuperIsGranted(permission);
+    }
+  }
+
+  void setNotificationGroupId(String value) {
+    _notificationGroupId = value;
+  }
+
+  Future<bool> callSuperIsGranted(Permission permission) {
+    return super.isGranted(permission);
+  }
+
+  Future<bool> checkGroupPermission() async {
+    if (_notificationGroupId.isEmpty) {
+      return false;
+    }
+
+    const methodChannel = MethodChannel('com.linagora.android.teammail.notification.group.permission');
```

Comment:
Should be only create one time

Reply:
![Screenshot 2024-06-07 at 16 43 11](https://github.com/linagora/tmail-flutter/assets/160106668/898eed23-36d0-43da-aee8-8e8ad6bb9516)

MethodChannel is constant. Creating multiple instance of it doesn't mean it takes more memory.

Reply:
Please attach official documents that prove what you say.

Reply:
[identical document](https://api.dart.dev/stable/2.6.1/dart-core/identical.html)